### PR TITLE
Ghost Orbit

### LINKED ISSUE #5437: Ghost Orbit
## Description
Ghost players should get the button to orbit because it is taxing to follow people around all the time. To make this work, the players pressing the existing orbit button down the bottom to follow another player from a menu and see their point of view. Roughly the following features should be implemented to achieve this:

- When the orbit button is pressed, the player initially selects from a menu similar to the “Teleport to Player” menu
- Their ghost sprite should orbit around the player that they are following in a circle.
- ghosts can orbit a number of different entities, not just players. They can orbit any object or item by double clicking it, allowing ghosts to follow singularities, the nuke disk, or other important things.
- The left and right movement keys allow them to cycle their viewpoint allowing them to quickly navigate around to other points of interest (probably would help admins out a lot)
- While in this mode the orbit button turns into a red X, notifying the player that by clicking it they will stop orbiting altogether and be free to move again. Also closes the menu that is similar to the “Teleport to Player” menu.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
Closes #5437 

Ghosts will follow any gameObject that has a pushPull compontent on it OR the singularity component and their sprites will rotate around them as well.
CL: [New] orbiting has been added. As a ghost, Double-click on an object to follow it and the orbit button will show you a list of all players you can orbit.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
Done, review pls

File: UnityProject/Assets/Scripts/Player/GhostOrbit.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		{
+			StopOrbiting();
+			UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+		}
+
+		public void Orbit(GameObject thingToOrbit)
+		{
+			target = thingToOrbit;
+			rotateTransform.TransformToRotateAround = thingToOrbit.transform;
+			netTransform.SetPosition(target.AssumedWorldPosServer(), false);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/GhostOrbit.cs:81]: This needs to be serverside

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Player/GhostOrbit.cs:81]: I think it's done now, review pls

**Review by Bod9001 — DISMISSED** [review]
waiting on dan to see if it's satisfactory

File: UnityProject/Assets/Scripts/Player/GhostOrbit.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				{
+					DoubleClickTimer();
+					return;
+				}
+				FindObjectToOrbitUnderMouse();
+			}
+			if(target == null) return;
+			if (KeyboardInputManager.IsMovementPressed())
+			{
+				StopOrbiting();
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/GhostOrbit.cs]: Needs to also be a command

**Comment by ThatDan123** [comment]
Also, can you test what happens when you are orbiting something and then use the teleport menu. Does it work ok?

**Comment by MaxIsJoe** [comment]
> Also, can you test what happens when you are orbiting something and then use the teleport menu. Does it work ok?

good catch, fixed it.

Ready for review

**Review by corp-0 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**